src/main/java/com/conveyal/gtfs/validator/PatternFinderValidator.java

```diff
@@ -72,19 +73,18 @@ public void validateTrip (Trip trip, Route route, List<StopTime> stopTimes, List
      */
     @Override
     public void complete(ValidationResult validationResult) {
-        Set<String> patternIds = new HashSet<>();
+        List<Pattern> patternsFromFeed = new ArrayList<>();
         for(Pattern pattern :  feed.patterns) {
-            patternIds.add(pattern.pattern_id);
+            patternsFromFeed.add(pattern);
         }
         LOG.info("Finding patterns...");
         Map<String, Stop> stopById = new HashMap<>();
         for (Stop stop : feed.stops) {
             stopById.put(stop.stop_id, stop);
         }
         // Although patterns may have already been loaded from file, the trip patterns are still required.
-        Map<TripPatternKey, Pattern> patterns = patternFinder.createPatternObjects(stopById, errorStorage);
-        patternBuilder.create(patterns, patternIds);
+        Map<TripPatternKey, Pattern> patterns = patternFinder.createPatternObjects(stopById, patternsFromFeed, errorStorage);
+        patternBuilder.create(patterns, !patternsFromFeed.isEmpty());
```

Comment:
Replace `!patternsFromFeed.isEmpty()` with `patternFinder.canUsePatternsFromFeed`. As written, I don't see any pattern renaming taking place either after importing a feed into a blank editor where ` canUsePatternsFromFeed` is false, or when viewing the "Patterns" tab in the feed version summary page.

Reply:
Fixed in: https://github.com/ibi-group/gtfs-lib/pull/6/commits/287badffe2860e94a320bcda62fee33213069541